libcloud/dns/drivers/worldwidedns.py

```diff
@@ -0,0 +1,506 @@
+        """
+        Return a list of zones.
+
+        :return: ``list`` of :class:`Zone`
+
+        For more info, please see:
+        https://www.worldwidedns.net/dns_api_protocol_list.asp
+        or
+        https://www.worldwidedns.net/dns_api_protocol_list_reseller.asp
+        """
+        action = '/api_dns_list.asp'
+        if self.reseller_id is not None:
+            action = '/api_dns_list_reseller.asp'
+        zones = self.connection.request(action)
+        if len(zones.body) == 0:
+            return []
+        else:
+            return self._to_zones(zones.body)
+
+    def list_records(self, zone):
+        """
+        Return a list of records for the provided zone.
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :return: ``list`` of :class:`Record`
+        """
+        records = self._to_records(zone)
+        return records
+
+    def iterate_records(self, zone):
+        """
+        Return a generator to iterate over records for the provided zone.
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :rtype: ``generator`` of :class:`Record`
+        """
+        records = self._to_records(zone)
+        for record in records:
+            yield record
+
+    def get_zone(self, zone_id):
+        """
+        Return a Zone instance.
+
+        :param zone_id: ID of the required zone
+        :type  zone_id: ``str``
+
+        :rtype: :class:`Zone`
+        """
+        zones = self.list_zones()
+        zone = [zone for zone in zones if zone.id == zone_id]
+        if len(zone) == 0:
+            raise ZoneDoesNotExistError(driver=self,
+                                        value="The zone doesn't exists",
+                                        zone_id=zone_id)
+        return zone[0]
+
+    def get_record(self, zone_id, record_id):
+        """
+        Return a Record instance.
+
+        :param zone_id: ID of the required zone
+        :type  zone_id: ``str``
+
+        :param record_id: ID of the required record
+        :type  record_id: ``str``
+
+        :rtype: :class:`Record`
+        """
+        zone = self.get_zone(zone_id)
+        records = self.list_records(zone)
+        record = [record for record in records if record.id == record_id]
+        if len(record) == 0:
+            raise RecordDoesNotExistError(value="Record doesn't exists",
+                                          driver=zone.driver,
+                                          record_id=record_id)
+        return record[0]
+
+    def update_zone(self, zone, domain, raw=False, type='master', ttl=None,
+                    extra=None):
+        """
+        Update en existing zone.
+
+        :param zone: Zone to update.
+        :type  zone: :class:`Zone`
+
+        :param domain: Zone domain name (e.g. example.com)
+        :type  domain: ``str``
+
+        :param raw: Mode we use to do the update using zone file or not.
```

Comment:
This is an extension (driver specific) argument so please prefix it with `ex_` (`ex_raw=False`) and put it at the end after `extra`.